### ISSUE TITLE
Implement error return on regression action

### ISSUE THREAD

**Issue by karkhaz** [model-checking/kani#2338]
Implement a benchcomp 'action' (previously called 'visualization') that terminates benchcomp with a return code of 1 if the result contains a regression.
This is required to be able to terminate CI runs when a regression is detected. It is also useful when running locally, for example when benchcomp is used as part of a makefile or script that checks whether a regression happened.
Users should be able to specify the action like this in their benchcomp.config file:
```yaml actions:
  - type: error_return_on_regression
    checks:
      - metric: passed
        check: boolean_not
      - metric: runtime
        check: ratio
        max: 1.1
```
This says: perform two actions on the results of running the two variants. Check if the 'passed' metric has flipped from true to false or vice versa; and check that the ratio of new to old runtime does not exceed 1.1.
If any pair of results fail either of these checks, the action sets the returncode of benchcomp to 1.
